packages/connect/src/device/Device.ts

```diff
@@ -547,6 +564,34 @@ export class Device extends TypedEmitter<DeviceEvents> {
         }
     }
 
+    async checkFirmwareRevision() {
+        const firmwareVersion = this.getVersion();
+
+        if (firmwareVersion.length === 0 || !this.features) {
+            return; // This happens when device has no features (not yet connected)
```

Comment:
I think this shouldn't actually happen at all. but I understand this check is needed because of typescript, right?